### DISCUSSION TITLE
Initial Setup Assistance

### DISCUSSION THREAD

**Discussion by lukedbarton** [jasonacox/Powerwall-Dashboard#77] [Q&A] [answered]
Hi All, I am kind of new to linux (ubuntu 22.04) (apologies in advance!) and grafana and probobally in the deep end and am looking for a little help to get this up and running cause it looks amazing and love metrics!

I have managed to get docker installed and it doesnt seem to be throwing any errors after following https://docs.docker.com/engine/install/ and configuring non sudo to access it.
I am now getting an issue after following the instructions to install compose successfully and then trying to run setup.sh

luke@tbcgfa01: /Powerwall-Dashboard$ ./setup.sh
Powerwall Dashboard (v2.5.2) - SETUP
ERROR: docker-compose is not available or not runnning.
This script requires docker-compose, please install and try again.

Compose its is def installed
luke@tbcgfa01:~/Powerwall-Dashboard$ docker compose version 
Docker Compose version v2.10.2

I have set the permission correct I think on tyhe compose file to allow non sudo to execute
sudo chmod +x /usr/libexec/docker/cli-plugins/docker-compose

(I had to use the find command to find this file after the auto install placed it here)
luke@tbcgfa01:~/Powerwall-Dashboard$ sudo find / -name "docker-compose" -type f
/usr/libexec/docker/cli-plugins/docker-compose

Any pointers would be great!

**Comment by jasonacox** [comment, marked as answer]
Thanks for posting this @lukedbarton - you might have identified an issue with the setup script!  It expects `docker-compose` to be available - can you run this and paste results?

```bash
# what setup uses
docker-compose version

# what you may have
docker compose version
```

I suspect you have v2 (the second one) but the setup script uses the v1 format.  Here is the way to install docker-compose:

```bash
# install
sudo pip3 install docker-compose

# test
docker-compose --version
```

**Reply by jasonacox** [reply]
I see the update.  Is docker-compose in your path so that you see it when you run `docker-compose version` from the command line that you get a response?  If not, it seem you need to add "/usr/libexec/docker/cli-plugins" to the path - something like this:

```bash
# update path
export PATH="/usr/libexec/docker/cli-plugins:$PATH"

# test
docker-compose version

# setup
./setup.sh
```

**Reply by lukedbarton** [reply]
Hey @jasonacox nope wasnt in my path! Thank you very much thats fixed it up for me for now! I will have another go this afternoon. Thank you very much for the response :-)

**Reply by mcbirse** [reply]
Hi @lukedbarton I might be able to help with this hopefully - Jason would definitely be sleeping right now considering our timezone difference! 😉 

The email and password in setup.sh is referring to the Tesla Gateway customer login credentials (not your Tesla account login).

If you have never configured the customer login password of the gateway, hopefully it is still set to the default.

This is listed in Step 7 of the guide below, and is dependant on your gateway version.
https://www.tesla.com/en_au/support/energy/powerwall/own/monitoring-from-home-network

Try logging in to the gateway via a normal web browser by going to `https://gateway-ip-address/` and entering the default password along with your email address. The login page should look like below:

![image](https://user-images.githubusercontent.com/108725631/190980980-af5033ff-3fc4-4a0a-a952-6c4e88b87712.png)

If it is not accepting the password, perhaps try the "Change or Forgot Password" link (I have never tried this personally so don't know if it works).

After a successful login you should see the power flow animation:

![image](https://user-images.githubusercontent.com/108725631/190982194-f00950cf-973e-4ad8-a66a-5b7bed355cce.png)

Once you have confirmed your gateway customer login password, you can run setup.sh again to update the config:
```
# setup
./setup.sh
```
Hope that helps!

**Reply by lukedbarton** [reply]
Thanks mate I just worked that out, feel like a bit of a rookie so took the question down.
Really appreciate the help!

**Reply by mcbirse** [reply]
No problem, glad you got it sorted!

**Reply by lukedbarton** [reply]
And just to close off thanks all for the assistance I got it up and running thank you again!

**Reply by lukedbarton** [reply]
Hey Team, my gateway has changed its ip address :-( and my new home opensource firewall doesnt support reservations. Any thoughts on if the gateway can be assigned a static IP or how can I modify my setup for the new IP address.
Cheers! Luke

**Reply by lukedbarton** [reply]
Should I just rerun the setup?

**Reply by jasonacox** [reply]
Hi @lukedbarton - yes you can re-run setup but there is an easier way:
1) Edit the `pypowerwall.env` file and change the IP address
2) Restart pypowerwall: `docker restart pypowerwall`

I'm surprised that your firewall doesn't allow IP assignments.  You might check on that again.
